<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>800</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Résolution de Labyrinthe - A* et Dijkstra</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QGroupBox" name="controlGroup">
      <property name="title">
       <string>Contrôles</string>
      </property>
      <property name="minimumWidth">
       <number>300</number>
      </property>
      <property name="maximumWidth">
       <number>350</number>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QGroupBox" name="dimensionsGroup">
         <property name="title">
          <string>Dimensions du Labyrinthe</string>
         </property>
         <layout class="QFormLayout" name="formLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="widthLabel">
            <property name="text">
             <string>Largeur:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="widthSpinBox">
            <property name="minimum">
             <number>5</number>
            </property>
            <property name="maximum">
             <number>50</number>
            </property>
            <property name="value">
             <number>15</number>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="heightLabel">
            <property name="text">
             <string>Hauteur:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="heightSpinBox">
            <property name="minimum">
             <number>5</number>
            </property>
            <property name="maximum">
             <number>50</number>
            </property>
            <property name="value">
             <number>15</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="obstaclesGroup">
         <property name="title">
          <string>Obstacles</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QLabel" name="obstacleTypeLabel">
            <property name="text">
             <string>Type d'obstacles:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="obstacleTypeCombo">
            <item>
             <property name="text">
              <string>Aléatoire</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Murs verticaux</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Murs horizontaux</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Motif complexe</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Aucun obstacle</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="densityLabel">
            <property name="text">
             <string>Densité (0.0 - 1.0):</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="densitySpinBox">
            <property name="maximum">
             <double>1.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.050000000000000</double>
            </property>
            <property name="value">
             <double>0.200000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="rewardsGroup">
         <property name="title">
          <string>Récompenses</string>
         </property>
         <layout class="QFormLayout" name="formLayout_2">
          <item row="0" column="0">
           <widget class="QLabel" name="stepCostLabel">
            <property name="text">
             <string>Coût par pas:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QDoubleSpinBox" name="stepCostSpinBox">
            <property name="minimum">
             <double>-100.000000000000000</double>
            </property>
            <property name="maximum">
             <double>0.000000000000000</double>
            </property>
            <property name="value">
             <double>-1.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="goalRewardLabel">
            <property name="text">
             <string>Récompense but:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QDoubleSpinBox" name="goalRewardSpinBox">
            <property name="maximum">
             <double>1000.000000000000000</double>
            </property>
            <property name="value">
             <double>100.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QCheckBox" name="bonusCheckBox">
            <property name="text">
             <string>Ajouter bonus:</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="numBonusLabel">
            <property name="text">
             <string>Nombre bonus:</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QSpinBox" name="numBonusSpinBox">
            <property name="minimum">
             <number>0</number>
            </property>
            <property name="maximum">
             <number>20</number>
            </property>
            <property name="value">
             <number>5</number>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="bonusValueLabel">
            <property name="text">
             <string>Valeur bonus:</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QDoubleSpinBox" name="bonusValueSpinBox">
            <property name="maximum">
             <double>100.000000000000000</double>
            </property>
            <property name="value">
             <double>10.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="generateButton">
         <property name="text">
          <string>Générer Labyrinthe</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="algorithmGroup">
         <property name="title">
          <string>Algorithmes</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QPushButton" name="solveAStarButton">
            <property name="text">
             <string>Résoudre avec A*</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="solveDijkstraButton">
            <property name="text">
             <string>Résoudre avec Dijkstra</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="compareButton">
            <property name="text">
             <string>Comparer les deux</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="clearButton">
         <property name="text">
          <string>Effacer le chemin</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QTextEdit" name="statsText">
         <property name="maximumHeight">
          <number>150</number>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGraphicsView" name="mazeView">
      <property name="renderHints">
       <set>QPainter::Antialiasing|QPainter::SmoothPixmapTransform|QPainter::TextAntialiasing</set>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1200</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFichier">
    <property name="title">
     <string>Fichier</string>
    </property>
    <addaction name="actionQuitter"/>
   </widget>
   <widget class="QMenu" name="menuAide">
    <property name="title">
     <string>Aide</string>
    </property>
    <addaction name="actionAPropos"/>
   </widget>
   <addaction name="menuFichier"/>
   <addaction name="menuAide"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionQuitter">
   <property name="text">
    <string>Quitter</string>
   </property>
  </action>
  <action name="actionAPropos">
   <property name="text">
    <string>À propos</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
